apps/aechannel/test/aesc_txs_SUITE.erl

```diff
@@ -2875,6 +2879,90 @@ fp_too_soon(Cfg) ->
     Test(100, 200, 100),
     ok.
 
+fp_register_name(Cfg) ->
+    Name = <<"lorem.test">>,
+    Salt = 42,
+    {ok, NameAscii} = aens_utils:to_ascii(Name),
+    CHash           = aeu_hex:hexstring_encode(
+                        aens_hash:commitment_hash(NameAscii, Salt)),
+    #{ public := Pubkey, secret := Privkey } = enacl:sign_keypair(),
+    Account = aeu_hex:hexstring_encode(Pubkey),
+    StateHashSize = aec_base58c:byte_size_for_type(state),
+    StateHash = <<42:StateHashSize/unit:8>>,
+    Round = 42,
+    FPRound = Round + 10,
+    Test =
+        fun(Owner, Forcer) ->
+            ContractCreateRound = 10,
+            run(#{cfg => Cfg},
+               [positive(fun create_channel_/2),
+                % store state on-chain via snapshot
+                set_from(initiator),
+                set_prop(round, 42),
+                set_prop(state_hash, StateHash),
+                positive(fun snapshot_solo_/2),
+                fun(#{state := S,
+                      channel_pubkey := ChannelPubKey} = Props) ->
+                    Channel = aesc_test_utils:get_channel(ChannelPubKey, S),
+                    Round = aesc_channels:round(Channel),
+                    StateHash = aesc_channels:state_hash(Channel),
+                    Props
+                end,
+                % create contract off-chain
+                create_trees_if_not_present(),
+                set_from(Owner, owner, owner_privkey),
+                create_contract_in_trees(_Round    = ContractCreateRound,
+                                         _Contract = "aens",
+                                         _InitArgs = <<"()">>,
+                                         _Deposit  = 2),
+                % force progress contract on-chain
+                fun(#{contract_id := ContractId} = Props) ->
+                    BinToSign = <<Pubkey/binary, ContractId/binary>>,
+                    <<Word1:256, Word2:256>> = enacl:sign_detached(BinToSign,
+                                                                   Privkey),
+                    Sig = aeu_hex:hexstring_encode(
+                            aeso_data:to_binary({Word1, Word2})),
+                    PreclaimArgs = <<"(", Account/binary, ",",
+                                          CHash/binary, ",",
+                                          Sig/binary,
+                                      ")">>,
+                    (force_call_contract_first(Forcer, <<"preclaim">>,
```

Comment:
It seems (I am not 100% sure, have not followed fully) that this calls `create_contract_call_payload` that calls `aesc_offchain_update:apply_on_trees`. That calls `aect_channel_contract:run` that calls the VM that [errs](https://35097-99802036-gh.circle-artifacts.com/1/home/builder/epoch/_build/test/logs/ct_run.epoch_ct@localhost.2018-10-08_16.08.20/lib.aechannel.aesc_txs_SUITE.logs/run.2018-10-08_16.11.29/aesc_txs_suite.fp_register_name.html) (not saying it is bad, just getting my head around it)
```
Return error error:{badmatch,
                       {error,
                           {badmatch,
                               <<0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
...
```
that, by enabling more stacktraces locally, I see it is on `aeso_data:heap_word(Heap,Addr=9039081679025068027825624833915204708939891774825472053301238029925278094596)` called by `aevm_eeevm_state:get_contract_call_input` at `{ok, Arg}  = aeso_data:from_heap(Type, Heap, Ptr)`. Is this the intended failure? (Getting my head around this.)

Reply:
It shouldn't be. _investigating_